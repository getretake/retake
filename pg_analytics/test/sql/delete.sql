CREATE TABLE employees (salary bigint, id smallint) USING deltalake;
INSERT INTO employees VALUES (100, 1), (200, 2), (300, 3), (400, 4), (500, 5);
DELETE FROM employees WHERE id = 5 OR salary <= 200;
SELECT * FROM employees;

DELETE FROM employees;

CREATE TABLE projects (project_id serial, employee_id int) using deltalake;
DELETE FROM employees
WHERE id NOT IN (SELECT employee_id FROM projects);

DROP TABLE employees, projects;
