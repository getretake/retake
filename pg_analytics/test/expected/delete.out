CREATE TABLE employees (salary bigint, id smallint) USING parquet;
INSERT INTO employees VALUES (100, 1), (200, 2), (300, 3), (400, 4), (500, 5);
DELETE FROM employees WHERE id = 5 OR salary <= 200;
SELECT * FROM employees;
 salary | id 
--------+----
    400 |  4
    300 |  3
(2 rows)

DELETE FROM employees;
ERROR:  Run TRUNCATE <table_name> to delete all rows from a table
CREATE TABLE projects (project_id serial, employee_id int) USING parquet;
DELETE FROM employees
WHERE id NOT IN (SELECT employee_id FROM projects);
ERROR:  Nested DELETE queries are not yet supported for parquet tables
DROP TABLE employees, projects;
